feat: Build hitpay app for cal.com

MuhammadAimanSulaiman opened the pull request:
Adding hitpay functions to hitpay app

kutsaniuk commented on packages/app-store/hitpay/api/add.ts:
```diff
@@ -0,0 +1,55 @@
+// import { createDefaultInstallation } from "@calcom/app-store/_utils/installation";
+// import type { AppDeclarativeHandler } from "@calcom/types/AppHandler";
+// import appConfig from "../config.json";
+// const handler: AppDeclarativeHandler = {
+//   appType: appConfig.type,
+//   variant: appConfig.variant,
+//   slug: appConfig.slug,
+//   supportsMultipleInstalls: false,
+//   handlerType: "add",
+//   createCredential: ({ appType, user, slug, teamId }) =>
+//     createDefaultInstallation({ appType, user: user, slug, key: {}, teamId }),
+// };
+// export default handler;
```
Hi @MuhammadAimanSulaiman, please remove.

kutsaniuk commented on packages/app-store/hitpay/api/callback.ts:
```diff
@@ -0,0 +1,39 @@
+// import type { Prisma } from "@prisma/client";
```
Please, remove it.

MuhammadAimanSulaiman replied:
I removed it

kutsaniuk commented on packages/app-store/hitpay/api/webhook.ts:
```diff
@@ -0,0 +1,61 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+// import getRawBody from "raw-body";
```
Please, remove.

MuhammadAimanSulaiman replied:
removed it

kutsaniuk commented on packages/app-store/hitpay/lib/PaymentService.ts:
```diff
@@ -0,0 +1,160 @@
+    bookerName: string,
+    paymentOption: PaymentOption,
+    bookerEmail: string
+    // bookerPhoneNumber?: string | null,
```
Please, remove unused code.

MuhammadAimanSulaiman replied:
I deleted all unused code

kutsaniuk commented on packages/app-store/hitpay/zod.ts:
```diff
@@ -0,0 +1,37 @@
+  value: z.string(),
+});
+
+export const paymentOptionsSchema = z.array(paymentOptionSchema);
```
If we don't need `paymentOptionsSchema`, please remove it.

MuhammadAimanSulaiman replied:
I removed it

kutsaniuk commented on packages/app-store/hitpay/static/1.jpeg:
Remove unused images

MuhammadAimanSulaiman replied:
The images are for the description on cal app store.
We shouldn't remove them and should replace them with ones related to hitpay.

kutsaniuk commented on packages/app-store/hitpay/pages/setup/index.tsx:
```diff
@@ -0,0 +1,265 @@
+  saltKey: string | null;
+}
+
+export interface IHitPayWebhookEventReturn {
```
Remove unused interface `IHitPayWebhookEventReturn`

MuhammadAimanSulaiman replied:
I removed it

kutsaniuk commented on packages/app-store/hitpay/components/HitpayPaymentComponent.tsx:
```diff
@@ -0,0 +1,43 @@
+import { useEffect } from "react";
+import { z } from "zod";
+
+// import type { PaymentPageProps } from "@calcom/features/ee/payments/pages/payment";
```
Remove it, please

MuhammadAimanSulaiman replied:
removed it

kutsaniuk commented on packages/app-store/hitpay/components/HitpayPaymentComponent.tsx:
```diff
@@ -0,0 +1,43 @@
+    // Will be parsed on render
+    data: unknown;
+  };
+  // paymentPageProps: PaymentPageProps;
```
Remove it, please

MuhammadAimanSulaiman replied:
removed it

kutsaniuk commented on packages/app-store/hitpay/components/HitpayPaymentComponent.tsx:
```diff
@@ -0,0 +1,43 @@
+
+export const HitpayPaymentComponent = (props: IPaymentComponentProps) => {
+  const { payment } = props;
+  console.log("HitpayPaymentComponent payment prop =>", payment);
```
Remove it, please

MuhammadAimanSulaiman replied:
I removed it

kutsaniuk commented on packages/app-store/hitpay/components/EventTypeAppSettingsInterface.tsx:
```diff
@@ -0,0 +1,113 @@
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Alert, Select, TextField } from "@calcom/ui";
+
+import { paymentOptions } from "../lib/constants";
```
Change with `import { paymentOptions } from "../lib";`

MuhammadAimanSulaiman replied:
I changed it

kutsaniuk commented on packages/app-store/hitpay/components/EventTypeAppCardInterface.tsx:
```diff
@@ -0,0 +1,50 @@
+  const { getAppData, setAppData, disabled } = useAppContextWithSchema<typeof appDataSchema>();
+  const { enabled, updateEnabled } = useIsAppEnabled(app);
+  const otherPaymentAppEnabled = checkForMultiplePaymentApps(eventTypeFormMetadata);
+  const [requirePayment, setRequirePayment] = useState(getAppData("enabled"));
```
Remove `setRequirePayment` if we don't need it.

MuhammadAimanSulaiman replied:
Removed it

MuhammadAimanSulaiman committed:
feat: Added defaultLink to hitpay setup

MuhammadAimanSulaiman committed:
feat: Added defaultLink to hitpay setup

kutsaniuk committed:
Update hitpay package.json

ThyMinimalDev committed:
chore: add TLD in username of managed users (#18229)

* chore: add TLD in username of managed users

* fixup! chore: add TLD in username of managed users

* bump platform libraries

supalarry committed:
feat: v2 wildcard redirectURIs (#18165)

* feat: isOriginAllowed function

* feat: use isOriginAllowed in v2

* refactor: shorten isOriginAllowed

anikdhabal committed:
fix: private team does not hide team members from manged event type (#18166)

Ryukemeister committed:
fix: `bookerEmbed` atom (#18168)

* update translations

* add custom hook for platform embed

* update event type ouput DTO response

* fixup

* display terms and conditions for embed

* update v2 docs

* PR feedback

* update docs

* PR feedback

* update docs

* update output DTO

* update docs

* update output typing

* docs update

* resolve merge conflicts

* merge conflicts resolving

* resolve merge conflicts

* fixup

* update docs

keithwillcode committed:
perf: More tracing for handleNewBooking (#18233)

supalarry committed:
feat: v2 teams CRUD (#18019)

* feat: teams endpoints

* refactor: organization teams reuse team types

* refactor: move team specific functions out of organizations-teams.repository

* supporting files

* regenerate docs

* feat: stripe service to generate team creation checkout

* refactor: request payment when creating a team

* fix: tests

* fix env

* driveby: event type name and email booking fields

* finish merge main

zomars committed:
feat: allows forcing/skipping calendar cache serving (#18224)

* feat: allows forcing/skipping calendar cache serving

Signed-off-by: Omar López <zomars@me.com>

* Update features.repository.ts

* Added to HNB

* type fixes

* Update packages/prisma/migrations/20241216000000_add_calendar_cache_serve/migration.sql

Co-authored-by: Keith Williams <keithwillcode@gmail.com>

* Update packages/prisma/zod-utils.ts

Co-authored-by: Keith Williams <keithwillcode@gmail.com>

* Update selectedCalendar.ts

---------

Signed-off-by: Omar López <zomars@me.com>
Co-authored-by: Keith Williams <keithwillcode@gmail.com>

Commit:
chore: release v4.8.2

hariombalhara committed:
fixes:CAL-4905 (#18222)

zomars committed:
fix: calendar cache date ranges (#18237)

Co-authored-by: Keith Williams <keithwillcode@gmail.com>

supalarry committed:
feat: v2 teams event types (#18085)

* feat: v2 teams event-types endpoints

* refactor: rely on teams event types service and repository

* finish mergin main into this branch

* test: teams event types

* refactor: move teams event types output from org to teams folder

* refactor: move teams event types output from org to teams folder

* chore: finish merge main

* refactor: logger name and unused import

* chore: finish main merge

---------

Co-authored-by: Somay Chauhan <somaychauhan98@gmail.com>

supalarry committed:
fix: v2 isSystemAdmin (#18162)

* refactor: ApiAuthStrategy return isSystemAdmin and define returned type

* refactor: use ApiAuthGuardUser instead of GetUserReturnType

* regenerate docs

nizzyabi committed:
feat: Plain integration (#18130)

* revert: "fix: Request permissions to allow events to be created on shared Office365/Outlook calendars (#17760)"

This reverts commit 1680cba5a55fb37d65ddd30d1e64b1451b5c6b31.

* plain.com card

* detiled error handling for get customer

* working email and id

* hmac

* Revert "fix: correct line-breaks in calendar event description (#18077)"

This reverts commit 06494a6999aa9e91087610ee2144d0a855d8395e.

* pr changes requet

* remove pan for now

* add-new-implementation-for-early-review

* Pushing fix for createHmac stringify

* added validation and user repository for email check

* add apiRouteMiddleware which handles the error handling

* HMAC_SECRET_KEY -> PLAIN_HMAC_SECRET_KEY

* Use the right error

* Convey right error to consumer

* Fixup apiRouteMiddleware to handle handler

* Don't export handler, only export POST

* changed to app directory

* working unkown user card

---------

Co-authored-by: Keith Williams <keithwillcode@gmail.com>
Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: Alex van Andel <me@alexvanandel.com>

zomars committed:
fix: single rejection caused cache cron to fail (#18239)

calcom-bot committed:
feat: update translations via @replexica (#18234)

* chore: sync with main

* feat: update translations via @replexica

---------

Co-authored-by: Replexica <support@replexica.com>

MuhammadAimanSulaiman committed:
fix: fix lint issue in KeyInput and update zod

sean-brydon committed:
feat: match edit user sheet designs (#18243)

* edit sheet design changes

* remove attribute header

anikdhabal committed:
fix: missing n8n app screenshot and add margin to the back button (#18240)

* fix: n8n app screenshot missing

* remove log

supalarry committed:
refactor: v2 calendars busy-times (#18244)

* refactor: make dateFrom and dateTo required

* refactor: make dateFrom and dateTo required

* refactor: useCalendarsBusyTimes enabled only if dateFrom & dateTo passed

supalarry committed:
fix: platform decrease usage when cancelling booking (#18241)

* fix: decrement billing upon cancelled booking

* docs: explain billing in FAQ

* update docs

hbjORbj committed:
chore: app router - /getting-started page (#18185)

* remove env vars

* move pages to /app

* remove Head component

* remove pages router

* add to config matcher

* fix

hbjORbj committed:
chore: app router - /reschedule pages (#18150)

* app router - /reschedule pages

* add to config.matcher

* generateMetdata is not needed

joeauyeung committed:
feat: Salesforce skip ownership check if attendee email is a free domain (#17916)

* Add email domain array

* Create numbered email domain object

* Check email domain

* Rename function

* Add tests

* Frontend enable skip ownership check if free email domain

* Backend ignore adding ownership to return records if free email domain check is enabled

* feat: Only require confirmation for free email domains (#17917)

* Add requiresConfirmationForFreeEmail to db

* Add option to event type settings

* Get requiresConfirmationForFreeEmail for event type page

* Include requiresConfirmationForFreeEmail in fetching event type

* Pass bookerEmail to `getRequiresConfirmationFlags`

* Add free email domain check to `determineRequiresConfirmation`

* Add `requiresConfirmationForFreeEmail` to types

* Add severity to Watchlist table

* Add migration for watchlist severity

* Add `getEmailDomainInWatchlist` method to watchlist repository

* Use watchlist repository to check for free email domain

* Mock watchlist repository in test

* Update test

* Rename method

* Add severity to blocked list

* Move check free email domain to async

* Type checks

* Adjust for promise returned

* Fix tests

* Fix

* Fix tests

emrysal committed:
chore: Set the stage for a primary key  with value uuid() (#18245)

* chore: Set the stage for a primary key  with value uuid()

* fix: Test and ensure it validates the id value is set and defined

* Add unique index manually, prisma doesn't support this

* revert unique

PeerRich committed:
Update i18n.json

CarinaWolli committed:
feat: ooo calibration (#17864)

* first version of ooo calibration

* decrease precision when checking for full day event

Co-authored-by: Alex van Andel <me@alexvanandel.com>

* improve variable naming

* don't include hosts created after OOO

* fix failing tests

* fix all tests

* add test for ooo calibration

* first implementation to fix full day busy event reaching the current month

* allow mutiple day events

* make getAvailabilityWithTimeZones optional

* fix type error

* fix type errors

* fix tests

* fix type error

* fix test

* code clean up

* remove time adjustment from map

* code clean up

* fix tests

* getAllCalendars -> this.getAllCalendars

* fix typo

* extract mergeOverlappingRanges

* code improvements from feedback

* fix getCalIdsWithTimeZone

---------

Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: Alex van Andel <me@alexvanandel.com>
Co-authored-by: Benny Joo <sldisek783@gmail.com>

sean-brydon committed:
refactor and fix bulk attribute handling (#18231)

zomars committed:
fix: skip failed cache subscriptions from retrying (#18252)

Commit:
chore: release v4.8.3

calcom-bot committed:
feat: update translations via @replexica (#18249)

* chore: sync with main

* feat: update translations via @replexica

---------

Co-authored-by: Replexica <support@replexica.com>

hbjORbj committed:
chore: app router - /teams single page (#18184)

* chore: app router - /teams single page

* fix type check

hbjORbj committed:
chore: bring back removed onboarding steps (#18254)

* add connected-calendar and connected-video steps back

* adapt test

* fix name

hbjORbj committed:
chore: app router - /booking pages (#18182)

* chore: app router - /booking pages

* move pages to /app

* add to config matcher

nizzyabi committed:
removed posthog identifier (#18253)

joeauyeung committed:
chore: CRM logging, add booking uid (#18256)

* Add more logging

* Add conditional

hbjORbj committed:
Revert "chore: app router - /reschedule pages (#18150)" (#18258)

This reverts commit a7f24e777b6876d96e70643be788d3a7d8cc80a7.

Commit:
chore: release v4.8.4

ibex088 committed:
fix: seats option not working for booker atom (#18265)

Ryukemeister committed:
fix: event type atom availability (#18263)

hbjORbj committed:
fix: include app directory in rewrites generation logic + migrate /reschedule pages to App Router (#18261)

* chore: app router - /reschedule pages (#18150)

* app router - /reschedule pages

* add to config.matcher

* generateMetdata is not needed

* include app directory in rewrites generation logic

* make it more readable

* make code more elegant

* fix

* fix

* fix

* fix

* test and refactor

Commit:
chore: release v4.8.5

supalarry committed:
fix: Booker atom default booking fields (#18275)

* refactor: rename transformBookingFieldsApiToInternal to transformBookingFieldsApiRequestToInternal

* refactor: default booking fields can required or not required

* fix: render system fields in booker

* fix: hidden property of booking fields

* fix: default field labels and placeholders

* chore: bump platform libraries

* fix: tests

* fix: e2e tests

ThyMinimalDev committed:
fix: platform booker icons and styling (#18276)

ThyMinimalDev committed:
fix: platform booker timezone select (#18277)

hariombalhara committed:
feat: Support event level conflict checking (#18052)

Co-authored-by: Alex van Andel <me@alexvanandel.com>
Co-authored-by: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>
Co-authored-by: Morgan Vernay <morgan@cal.com>
Co-authored-by: Omar López <zomars@me.com>

calcom-bot committed:
feat: update translations via @replexica (#18279)

Co-authored-by: Replexica <support@replexica.com>

Commit:
chore: release v4.8.6

zomars committed:
fix: prevents cron to be clogged with calendars with missing credentialId (#18281)

ThyMinimalDev committed:
chore: add cal.com logo to platform booker embed (#18269)

* chore: add cal.com logo to platform booker embed

* fixup! chore: add cal.com logo to platform booker embed

---------

Co-authored-by: Rajiv Sahal <sahalrajiv-extc@atharvacoe.ac.in>

retrogtx committed:
fix portugese (pt-BR) password validation from a required eight to seven. (#18273)

mintlify[bot] committed:
Mintlify Docs Edits (#18301)

Co-authored-by: mintlify[bot] <109931778+mintlify[bot]@users.noreply.github.com>

PeerRich committed:
feat: added chatbase app (#18302)

hbjORbj committed:
chore: add cache to static pages in app router (#18278)

* add cache

* fix

* refactor

hbjORbj committed:
fix: make sure to call refresh for pages rendered by app router (#18304)

ThyMinimalDev committed:
fix: hotfix hide hidden booking fields in event-types GET /  (#18306)

Commit:
chore: release v4.8.7

anikdhabal committed:
fix: Reschedule email for RR event shows wrong timezone to attendee (#18262)

PeerRich committed:
Create getting-started.mdx (#18342)

calcom-bot committed:
Delete help directory (#18343)

PeerRich committed:
chore: moved docs/help to /help (#18345)

PeerRich committed:
chore: Delete unused guides directory (#18346)

Udit-takkar committed:
feat: booking filters (#18303)

Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>

hbjORbj committed:
chore: Remove `HeadSeo` components where no longer needed + improve app router metadata logic (#18348)

* remove HeadSeo for already migrated pages and refactor prepareMetadata

* create _generateMetadataWithoutImage and refactor _generateMetadata

hbjORbj committed:
chore: app router - /bookings status page (#18183)

* chore: app router - /bookings page

* remove env vars

* fix

* Update middleware.ts

* revert unneeded change

* refactor for the better

* fix

hbjORbj committed:
cache i18n instances (#18309)

CarinaWolli committed:
feat: virtual queues tab in insights (#18260)

Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: sean-brydon <sean@cal.com>
Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Omar López <zomars@me.com>

calcom-bot committed:
feat: update translations via @replexica (#18361)

Co-authored-by: Replexica <support@replexica.com>

vijayraghav-io committed:
update OOO e2e tests to remove flakiness (#18367)

anikdhabal committed:
fix: metadata is overwritten for child managed eventType when updating parent (#18059)

* fix: metadata is overwirten

* Update

* type error

* Update

* fix test

* fix type error

PeerRich committed:
chore: added routing support link (#18369)

* Added routing form support link

* small change

* Type fix

---------

Co-authored-by: Joe Au-Yeung <j.auyeung419@gmail.com>

hbjORbj committed:
chore: refactor handling logic for embeds in app router (#18362)

* refactor handling logic for embeds in app router

* fix type checks

* add test for withEmbedSsrAppDir

* fix

nizzyabi committed:
feat: Plain chat (#18284)

* plain custom desin

* No display on card

* dynamic plain chat component and hmac hash

* re-route users to plain.com chat instead of intercom chat when going to /support

* provider errors

* yarn lock fix

* plain chat removed unneeded hmac

* fix ts error

* error handling improved

* remove intercome provider from app-dir

* Create getting-started.mdx (#18342)

* Delete help directory (#18343)

* chore: moved docs/help to /help (#18345)

* chore: Delete unused guides directory (#18346)

* feat: booking filters (#18303)

Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>

* chore: Remove `HeadSeo` components where no longer needed + improve app router metadata logic (#18348)

* remove HeadSeo for already migrated pages and refactor prepareMetadata

* create _generateMetadataWithoutImage and refactor _generateMetadata

* chore: app router - /bookings status page (#18183)

* chore: app router - /bookings page

* remove env vars

* fix

* Update middleware.ts

* revert unneeded change

* refactor for the better

* fix

* cache i18n instances (#18309)

* feat: virtual queues tab in insights (#18260)

Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: sean-brydon <sean@cal.com>
Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Omar López <zomars@me.com>

* feat: update translations via @replexica (#18361)

Co-authored-by: Replexica <support@replexica.com>

* update OOO e2e tests to remove flakiness (#18367)

* fix: metadata is overwritten for child managed eventType when updating parent (#18059)

* fix: metadata is overwirten

* Update

* type error

* Update

* fix test

* fix type error

* chore: added routing support link (#18369)

* Added routing form support link

* small change

* Type fix

---------

Co-authored-by: Joe Au-Yeung <j.auyeung419@gmail.com>

* chore: refactor handling logic for embeds in app router (#18362)

* refactor handling logic for embeds in app router

* fix type checks

* add test for withEmbedSsrAppDir

* fix

* review changes

* yarn lock

---------

Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Calcom Bot <109866826+calcom-bot@users.noreply.github.com>
Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>
Co-authored-by: Benny Joo <sldisek783@gmail.com>
Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>
Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: sean-brydon <sean@cal.com>
Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>
Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: Replexica <support@replexica.com>
Co-authored-by: Vijay <vijayraghav22@gmail.com>
Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>
Co-authored-by: Joe Au-Yeung <j.auyeung419@gmail.com>
Co-authored-by: Alex van Andel <me@alexvanandel.com>

asadath1395 committed:
fix: discrepancy in description hyperlinks and calendar invite (#18365)

anikdhabal committed:
fix: Recurring meeting creates increasing number of Zoom events (#18360)

asadath1395 committed:
fix: Last used label in button shown in 2-factor code and backup screen (#18341)

kart1ka committed:
feat: add name and email url params when rescheduling past bookings (#18318)

* feat: add name and email as url params when rescheduling past bookings

* remove logs

* pass booking seat name and email as url params

alishaz-polymath committed:
fix: update contact email for CalDAV (#18351)

Commit:
chore: release v4.8.8

nizzyabi committed:
remove yarnlock (#18377)

MuhammadAimanSulaiman committed:
fix: fix issue of parsing string to number

hariombalhara committed:
perf: /router endpoint perf improvement (#18366)

hariombalhara committed:
Fix routing form field input loosing focus on typing (#18381)

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

emrysal committed:
chore: rely less on autoinfer, be able to use initial data types in consumers (#18359)

Udit-takkar committed:
perf: getAvailableSlots improvements (#18382)

invalid-email-address committed:
chore: release v4.8.9

hariombalhara committed:
fix: Handle more dry-run cases (#18286)

Udit-takkar committed:
perf: improve checkIsAvailable in getAvailable slots (#18352)

* refactor: Update booking query to use userId and optimize availability check logic

- Changed the booking query to use userId instead of nested user object for filtering active bookings.
- Refactored availability check logic to improve readability and performance by using valueOf() for time comparisons and consolidating multiple checks into a single return statement.

* fix: update tests

* chore: use toDate()

* tests: add checkavailable unit tests

calcom-bot committed:
feat: update translations via @replexica (#18389)

* chore: sync with main

* feat: update translations via @replexica

---------

Co-authored-by: Replexica <support@replexica.com>

nizzyabi committed:
fix: removed plain chat from app.cal.com/video (#18385)

* fix: removed plain chat from app.cal.com/video

* take user session and open plain out of use effect

* move undefined check and screen size check above useeffect

* turned restrcited paths into env var

* undefeined window below hooks

* chore: add window check

* fix: infinite re rendering

* remove console.log

---------

Co-authored-by: Udit Takkar <udit222001@gmail.com>

anikdhabal committed:
fix: localized date (#18388)

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

Udit-takkar committed:
refactor: email handler (#18391)

Co-authored-by: Bailey Pumfleet <bailey@pumfleet.co.uk>

invalid-email-address committed:
chore: release v4.8.10

devin-ai-integration[bot] committed:
fix: reduce mobile navigation bottom margin from 0.625rem to 0.25rem (#18398)

nizzyabi committed:
fix: embed showing chat (#18404)

* fix: embed showing chat

* change url check to starts with

* hashmap through pathname

anikdhabal committed:
Revert "update OOO e2e tests to remove flakiness (#18367)" (#18411)

This reverts commit e6ef4e19881fe89ff10e92f263f1d7416be4cca2.

ShashankGupta10 committed:
fix: add typeform logo and fix alt text in /apps/typeform/how-to-use (#18393)

* :truck:: add typeform logo and fix alt text

* trying to fix the test

* Update testUtils.ts

---------

Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>
Co-authored-by: unknown <adhabal2002@gmail.com>

hbjORbj committed:
chore: reduce test time for onboarding e2e test (#18403)

vijayraghav-io committed:
fix: flaky OOO tests without time dependency (#18412)

* fix flaky OOO tests without time dependency

* chore

invalid-email-address committed:
chore: release v4.8.11

anikdhabal committed:
fix: Error changing location for meeting (#18401)

cnhhoang850 committed:
Fix duplicated data returned by Prisma query in team availability view (#18416)

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

cnhhoang850 committed:
Fix duplicated data returned by Prisma query in team availability view (#18416)

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

hbjORbj committed:
chore: Migrate /403 and /500 pages to App Router (#18424)

* global-error is not needed

* remove /403 and /500 pages from pages router

* add 403 and 500 pages in App Router

* add i18n strings

* fix i18n string

* fix styling

hbjORbj committed:
chore: Migrate /403 and /500 pages to App Router (#18424)

* global-error is not needed

* remove /403 and /500 pages from pages router

* add 403 and 500 pages in App Router

* add i18n strings

* fix i18n string

* fix styling

emrysal committed:
fix: Some recently discovered type errors breaking build (not sure why undetected before) (#18432)

emrysal committed:
fix: Some recently discovered type errors breaking build (not sure why undetected before) (#18432)

anikdhabal committed:
fix: Add to google leads to a 403 page (#18430)

anikdhabal committed:
fix: Add to google leads to a 403 page (#18430)

hariombalhara committed:
feat: Add dry run mode functionality and message display in booking process (#18429)

- Introduced a new `DryRunMessage` component to inform users when in dry run mode.
- Implemented utility function `isBookingDryRun` to check for dry run status based on URL search parameters.
- Updated the `Booker` component to conditionally render the dry run message.
- Adjusted booking mutation input mapping to reuse the utility

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

hariombalhara committed:
feat: Add dry run mode functionality and message display in booking process (#18429)

- Introduced a new `DryRunMessage` component to inform users when in dry run mode.
- Implemented utility function `isBookingDryRun` to check for dry run status based on URL search parameters.
- Updated the `Booker` component to conditionally render the dry run message.
- Adjusted booking mutation input mapping to reuse the utility

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

ThyMinimalDev committed:
fix: atoms globals.css invalid class format (#18295)

ThyMinimalDev committed:
fix: atoms globals.css invalid class format (#18295)

hbjORbj committed:
chore: migrate /icons page to App Router (#18436)

* migrate icons page

* migrate main index page

* remove /icons page

* use i18n string

* fix imports

* add icons_showcase i18n string

* add /app dir to tailwind preset config

* use classname

* do not migrate main index page

hbjORbj committed:
chore: migrate /icons page to App Router (#18436)

* migrate icons page

* migrate main index page

* remove /icons page

* use i18n string

* fix imports

* add icons_showcase i18n string

* add /app dir to tailwind preset config

* use classname

* do not migrate main index page

supalarry committed:
fix: booker atom booking fields (#18441)

supalarry committed:
fix: booker atom booking fields (#18441)

eunjae-lee committed:
feat: apply full width automatically for DataTable (#18357)

* feat: apply full width automatically for DataTable

* change implementation

* load all columns of insights routing table at the same time

* update team member list

* sticky columns for >= sm

* fix type error

---------

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

eunjae-lee committed:
feat: apply full width automatically for DataTable (#18357)

* feat: apply full width automatically for DataTable

* change implementation

* load all columns of insights routing table at the same time

* update team member list

* sticky columns for >= sm

* fix type error

---------

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

calcom-bot committed:
feat: update translations via @replexica (#18438)

* chore: sync with main

* feat: update translations via @replexica

---------

Co-authored-by: Replexica <support@replexica.com>

calcom-bot committed:
feat: update translations via @replexica (#18438)

* chore: sync with main

* feat: update translations via @replexica

---------

Co-authored-by: Replexica <support@replexica.com>

Udit-takkar committed:
fix: get bookings ordering (#18439)

* fix: get bookings ordering

* chore: update query

* chore: add log for debuggin on prod

* chore

Udit-takkar committed:
fix: get bookings ordering (#18439)

* fix: get bookings ordering

* chore: update query

* chore: add log for debuggin on prod

* chore

cnhhoang850 committed:
fix: improve useInViewObserver hook to trigger when observer is already in view on mount (#18446)

cnhhoang850 committed:
fix: improve useInViewObserver hook to trigger when observer is already in view on mount (#18446)

retrogtx committed:
fix: react incompatiblity with nextjs error (#18433)

retrogtx committed:
fix: react incompatiblity with nextjs error (#18433)

anikdhabal committed:
fix: Zoho Bigin not pulling information from booking (#18445)

* fix: Zoho Bigin not pulling information from booking

* update

* missed

* fix contactcreation

anikdhabal committed:
fix: Zoho Bigin not pulling information from booking (#18445)

* fix: Zoho Bigin not pulling information from booking

* update

* missed

* fix contactcreation

hariombalhara committed:
fix: Enhance SCIM payload handling and attribute creation (#18427)

* feat: Enhance SCIM user attribute handling and sync process

- Introduced a new PR_TODO.md file to track ongoing tasks and fixes.
- Updated `getAttributesFromScimPayload` to accept a directoryId and ignore core user attributes during syncing.
- Modified `handleUserEvents` to pass directoryId when syncing custom attributes.
- Improved attribute creation logic to handle unique options and added support for optional isGroup property in attribute options.
- Added utility function `getOptionsWithValidContains` to ensure valid sub-options in attribute options.
- Updated tests to reflect changes in attribute handling and ensure proper functionality.

This commit addresses issues with user attribute syncing and enhances the overall attribute management process.

* test: Add unit tests for getOptionsWithValidContains utility function

- Introduced a new test file for the getOptionsWithValidContains function.
- Added tests to verify the removal of duplicate options, initialization of empty contains arrays, filtering of non-existent sub-options, and handling of empty options arrays.
- Ensured comprehensive coverage of the utility's functionality to enhance reliability and maintainability.

hariombalhara committed:
fix: Enhance SCIM payload handling and attribute creation (#18427)

* feat: Enhance SCIM user attribute handling and sync process

- Introduced a new PR_TODO.md file to track ongoing tasks and fixes.
- Updated `getAttributesFromScimPayload` to accept a directoryId and ignore core user attributes during syncing.
- Modified `handleUserEvents` to pass directoryId when syncing custom attributes.
- Improved attribute creation logic to handle unique options and added support for optional isGroup property in attribute options.
- Added utility function `getOptionsWithValidContains` to ensure valid sub-options in attribute options.
- Updated tests to reflect changes in attribute handling and ensure proper functionality.

This commit addresses issues with user attribute syncing and enhances the overall attribute management process.

* test: Add unit tests for getOptionsWithValidContains utility function

- Introduced a new test file for the getOptionsWithValidContains function.
- Added tests to verify the removal of duplicate options, initialization of empty contains arrays, filtering of non-existent sub-options, and handling of empty options arrays.
- Ensured comprehensive coverage of the utility's functionality to enhance reliability and maintainability.

acharyayush committed:
fix: Timezone in Email Embed not working #15525 (#15537)

Co-authored-by: Keith Williams <keithwillcode@gmail.com>
Co-authored-by: Amit Sharma <74371312+Amit91848@users.noreply.github.com>
Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>
Co-authored-by: Hariom <hariombalhara@gmail.com>

eunjae-lee committed:
feat: server side sorting for insights routing (#18113)

* feat: server side sorting for insights routing

^ Conflicts:
^	packages/prisma/playground.ts

* fix download csv

* fix hooks locations

joeauyeung committed:
chore: Salesforce - Add additional logging and error checks (#18448)

cnhhoang850 committed:
fix: improve useInViewObserver hook to trigger when observer is already in view on mount (#18455)

alishaz-polymath committed:
fix: Show managed event bookings for team admins/owners (#18434)

* add managed event bookings for team admins/owners

* remove unnecessary OR

* address change suggestion

* improve readability

invalid-email-address committed:
chore: release v4.8.12

hbjORbj committed:
chore: app router - /team, /org, /[user] booking pages (excl. embeds) (#18186)

* update env vars

* update middleware

* remove pages router and move pages to /app

* move to /team

* update imports

* fix

* remove pages router and move org pages to /app

* wip

* fix orgSlug/user pages

* fix orgSlug/user/type pages

* remove generateMetadata from embed pages

* fix

* remove pages router for /user pages

* generateMetadata is not needed in embed pages

* remove future/org

* no layout in [user] page

* simplify

* fix

* fix OG image for [user]

* fix OG images for org/user and team/slug

* fix OG images for booking page

* fix all metadata

* use isBrandingHidden

* remove BookerSeo and its usages

* rename excludeAppNameFromTitle -> hideBranding

* remove logic for meeting type in HeadSeo

* remove BookerSeo instances from team-view and users-public-view

* create generateMeetingMetadata util and use it to reduce duplicate code

* remove BookerSeo imports

* fix spacing

* remove constructMeetingImage mock from head-seo.test

* fetch avatarUrl using user id for user page metadata

* fix test

* remove unused test cases

* index and follow must be true by default

* invert noindex/nofollow flags

* remove HeadSeo for already migrated pages and refactor prepareMetadata

* fix organization-settings.e2e.ts

* fix order

* enable parallel test execution for dynamic-booking e2e test

* fix

* + could be %2B in app router

* refactor handling logic for embeds in app router

* fix isEmbed

* fix embed-core

* remove dead code

* move embed pages back to /future

* add back embed pages in pages router

* revert some changes

* fix import type checks

* simplify

* fix

* feat: Implement generateBookingPageMetadata function for improved SEO and metadata handling across user and event pages (#18440)

- Added a new utility function `generateBookingPageMetadata` to streamline the generation of metadata for booking and user profile pages.
- Updated multiple page components to utilize the new function, enhancing SEO indexing and metadata consistency.
- Removed redundant code related to previous metadata generation methods, improving code clarity and maintainability.

* fix dirs

* Pr-review-fixes-app-router-team-pages (#18450)

* Remove unnecessary getPublicEvent call

* Remove unused variable

* Fix 404 for team page (#18451)

* Remove console log

---------

Co-authored-by: Hariom Balhara <hariombalhara@gmail.com>

hbjORbj committed:
chore: app router - /signup page (#18151)

* app router - /signup page

* add to config.matcher

* fix

reneaaron committed:
fix: improve qr readability of alby payment integration (#18266)

* fix: optimize qr for better readability

* fix: optimize readability

Amit91848 committed:
fix: It's hard to click on the last available Event Duration (#18462)

emrysal committed:
perf: Use improved checkIfAvailable and remove dupe code (#18463)

Amit91848 committed:
feat: Info lost when navigating without warning routing form (#18386)

Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

ThyMinimalDev committed:
docs: cal-api-version header example for bookings controller v2 (#18271)

* docs: api v2 cal-api-version header example for bookings controller

* fixup! docs: api v2 cal-api-version header example for bookings controller

Amit91848 committed:
feat: refund policies for payment apps (#18428)

* feat: refund policies for payment apps

* tests for payment refund

---------

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

Akashjayan1999 committed:
fix: initialize useCustomTemplate state based on payloadTemplate prop… (#18257)

kart1ka committed:
feat: add cancel and reschedule reason variables in workflows (#18305)

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

supalarry committed:
docs: OAuth client redirectUri with wildcard character (#18267)

VK-RED committed:
feat: shorten sms links using dub (#17925)

* feat: add getshortenLinks function

* feat: use shortenLinks in SMS Worflows

* chore: use getShortenLink and cleanup getShortenLinks

* fix: Duplicate keys error

---------

Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Alex van Andel <me@alexvanandel.com>

hariombalhara committed:
chore: Bump version for embed (#18285)

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

calcom-bot committed:
feat: update translations via @replexica (#18472)

* chore: sync with main

* feat: update translations via @replexica

---------

Co-authored-by: Replexica <support@replexica.com>

ThyMinimalDev committed:
chore: ignore oAuthClientId for Booker Embed bookings (#18314)

* chore: ignore oAuthClientId for Booker Embed bookings

* remove logs

* chore: isEmbed header and emails

* fixup! chore: isEmbed header and emails

* fixup! fixup! chore: isEmbed header and emails

* fixup! fixup! fixup! chore: isEmbed header and emails

* fixup! fixup! fixup! fixup! chore: isEmbed header and emails

* fixup! fixup! fixup! fixup! fixup! chore: isEmbed header and emails

---------

Co-authored-by: Ryukemeister <sahalrajiv6900@gmail.com>
Co-authored-by: Rajiv Sahal <sahalrajiv-extc@atharvacoe.ac.in>

supalarry committed:
refactor: v2 default booking fields (#18287)

* feat: allow title,notes,reschedule reason inputs

* refactor: v2 event type default booking fields as in web app

* wip: input validation

* input field validation

* fix: tests

* chore: bump libraries

* remove change in docs

* fix input type

* chore: ensure additional notes at the end

* fixup! Merge branch 'main' into lauris/cal-4932-platform-refactor-default-booking-fields

---------

Co-authored-by: Morgan Vernay <morgan@cal.com>
Co-authored-by: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>

anikdhabal committed:
fix: Zoho Calendar is allowing bookings during busy schedules (#18473)

* fix: zoho calendar allowing to book busy schedule

* Improved more readable version

---------

Co-authored-by: Alex van Andel <me@alexvanandel.com>

invalid-email-address committed:
chore: release v4.8.13

hariombalhara committed:
feat: Single domain setup when only one org in the system (#18383)

Co-authored-by: Omar López <zomars@me.com>

Praashh committed:
fix: Cannot read properties of undefined (reading 'timeZone') (#18107)

Co-authored-by: Hariom <hariombalhara@gmail.com>

VK-RED committed:
fix: cancel workflows on cancel booking (#17871)

* feat: cancel scheduled email from sendGrid while cancelling the booking

* fix: remove deleteScheduledSend in scheduledEmailReminder cron api

* fix: use cancelScheduledEmailSend

* chore: remove cancelling email workflow in deleteScheduledEmailReminder

* refactor: bring back deleteScheduleSend

* fix: get only past reminders

hbjORbj committed:
insights layout (#18489)

supalarry committed:
fix: v2 slots timeZone parameter (#18488)

* fix: v2 slots timeZone parameter

* test: v2 slots

hbjORbj committed:
withEmbedSsr is missing in a embed route (#18483)

hbjORbj committed:
feat: Prevent re-renders (moving SessionProvider to Root level for both app router / pages router) (#18493)

emrysal committed:
perf: Remove zod from edge runtime (#18494)

hbjORbj committed:
chore: remove unused code from SideBar (#18487)

zomars committed:
Update repository.ts (#18482)

hariombalhara committed:
fix: Make it easier to manage group options and also improve attribute options performance with 100s of options (#18486)

hbjORbj committed:
feat: bookings - render layout only once (#18490)

hbjORbj committed:
remove MainLayout / MainLayoutAppDir (#18498)

hariombalhara committed:
fix: 404 on the public form page for a sub-team form when the owner itself was removed from the org (#18495)

* wip

* fix: 404 for a form of a member that has been removed.

- Moved existing authorization check to isAuthorizedToViewFormOnOrgDomain function to handle form access control
- Add comprehensive test suite for org domain authorization scenarios
- Refactor getServerSideProps to use the new authorization function
- Improve code organization by separating domain-specific authorization logic

Test Coverage:
- Non-org domain access
- Org member's form access
- Sub-team form access within org
- Access denial for non-org members/teams

emrysal committed:
chore: Remove all instances of useIntercom (#18500)

hariombalhara committed:
fix checklyt tests (#18501)

Udit-takkar committed:
Revert "fix: improve useInViewObserver hook to trigger when observer is alrea…" (#18504)

This reverts commit 22f92f1ee3d9d734aef6657679591254dd06a603.

emrysal committed:
fix: type + build errors (#18502)

invalid-email-address committed:
chore: release v4.8.14

calcom-bot committed:
feat: update translations via @replexica (#18496)

Co-authored-by: Replexica <support@replexica.com>

joeauyeung committed:
chore: Add Salesforce tests to `getContacts` method (#18225)

hbjORbj committed:
chore: app router - /event-types pages (#18181)

* refactor event type web wrapper

* finish migration

* move event-types folder out of future

* remove vars

* remove pages router

* fix

* fix

* fix test

* remove test tag

* refactor

* remove completely

* add withoutSeo prop for event type web page

* remove HeadSeo usage from event types listing view

* make onboarding e2e test faster

* remove useIntercom

---------

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

hbjORbj committed:
perf: use react svg cache from root level (#18506)

eunjae-lee committed:
feat: add DataTableWrapper (#18503)

* feat: data table wrapper

* feat: add DataTableWrapper

---------

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

eunjae-lee committed:
fix: add index to AssignmentReason (#18520)

hbjORbj committed:
chore: Revert "chore: app router - /team, /org, /[user] booking pages (excl. embeds) (#18186) (#18526)

7hakurg committed:
feat-insihts-app (#18461)

zomars committed:
feat: Add auto payment methods for paid booking on Stripe (#18513)

asadath1395 committed:
fix: Description not rendered as html in select event types screen when installing an app (#18354)

Udit-takkar committed:
feat: support phone based booking api (#17635)

* chore: save progress

* feat: support phone based booking using API v2

* chore: change name of the variable

* fix: update event types and booking fields for improved validation and new properties

- Updated imports in input and output event types services to use the correct library version.
- Added new `avatarUrl` and `successRedirectUrl` properties to the Swagger documentation for user and booking schemas.
- Enhanced booking fields validation by changing the `required` property to a boolean type.
- Improved descriptions and examples in the OpenAPI specifications for better clarity.
- Refactored booking fields input and output classes to include new properties and ensure proper validation.

* fix: booking with phone number

* test: add event type update test

* test: add test for creating booking

* chore: update unit test

* fix: tests

* fix: add turbo

* chore: remove emial

* refactor: make email optional and move it to service

* fix: test

* chore: bump platform-libraries

* fix: e2e test

---------

Co-authored-by: supalarry <laurisskraucis@gmail.com>

ThyMinimalDev committed:
feat: api v2 org ooo crud (#18499)

* feat: api v2 org ooo crud

* chore: add guard isUserOOO on ooo controller

* chore: add checks for ooo

* fixup! chore: add checks for ooo

* added e2e tests

* fixup! added e2e tests

invalid-email-address committed:
chore: release v4.8.15

PeerRich committed:
chore: moved docs into mintlify (#18536)

MuhammadAimanSulaiman committed:
fix: add default hitpay api links

Udit-takkar committed:
chore: change identifier (#18518)

Co-authored-by: Bailey Pumfleet <bailey@pumfleet.co.uk>

ibex088 committed:
fix: payment not working in booker (#18540)

* fix: payment not working in booker

* Update transformApiEventTypeForAtom.ts

---------

Co-authored-by: Rajiv Sahal <sahalrajiv-extc@atharvacoe.ac.in>

MuhammadAimanSulaiman committed:
feat: add event name as purpose to payment request api

mintlify[bot] committed:
Documentation edits made through Mintlify web editor (#18543)

PeerRich committed:
chore: fixed help logo (#18551)

PeerRich committed:
chore: mintlify updates (#18552)

PeerRich committed:
Documentation edits made through Mintlify web editor

PeerRich committed:
Documentation edits made through Mintlify web editor

PeerRich committed:
Documentation edits made through Mintlify web editor

PeerRich committed:
Documentation edits made through Mintlify web editor

PeerRich committed:
Documentation edits made through Mintlify web editor

mintlify[bot] committed:
feat: Mintlify /help edits (#18555)

* Documentation edits made through Mintlify web editor

* Documentation edits made through Mintlify web editor

* Documentation edits made through Mintlify web editor

* Documentation edits made through Mintlify web editor

---------

Co-authored-by: mintlify[bot] <109931778+mintlify[bot]@users.noreply.github.com>

nizzyabi committed:
fix: added videos and fixed iframe sizes for help page docs (#18558)

anikdhabal committed:
chore: temporarily skip a failing E2E test (#18560)

hbjORbj committed:
chore: unique key missing for child (#18542)

sean-brydon committed:
fix: Handle booker UI busy or skip on timeslot (#18553)

alishaz-polymath committed:
Update README.md (#18524)

keithwillcode committed:
perf: Improve routing insights view (#18557)

* perf: Improve routing insights view

* fix: remove old join

---------

Co-authored-by: Eunjae Lee <hey@eunjae.dev>
Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

hbjORbj committed:
chore: clean up some code in app router (#18541)

Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

mintlify[bot] committed:
Documentation edits made through Mintlify web editor (#18565)

Co-authored-by: mintlify[bot] <109931778+mintlify[bot]@users.noreply.github.com>

mintlify[bot] committed:
mintlify help edits (#18562)

* Documentation edits made through Mintlify web editor

* Documentation edits made through Mintlify web editor

* Delete help/calendars/how-to-add-apple-calendar.mdx

* Delete help/calendars/how-to-add-google-calendar.mdx

---------

Co-authored-by: mintlify[bot] <109931778+mintlify[bot]@users.noreply.github.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>

PeerRich committed:
Documentation edits made through Mintlify web editor

joeauyeung committed:
chore: Salesforce - optimize `getContact` method [CAL-4940] (#18289)

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

steven-tey committed:
feat: bulk shorten links with dub.links.createMany (#18539)

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

joeauyeung committed:
chore: Upgrade `jsforce` to 3.6.2 and use `jsforce-node` (#18363)

* Upgrade jsforce to 3.6.2

* Refactor connecting to Salesforce

* Revert yarn.lock changes

* Migrate callback endpoint

* Add jsforce node dependency

* Migrate add endpoint

* Import

* Import package into crmService

* Use new package types

* Type fix

* Update vite config

* Push updated lockfile

* Attempt to bump platform/libraries to unlock jsforce

* Also update lockfile, naturally

* bump platform libraries

* Update vite.config

* Add jsforce to vite config

* fixup! Merge branch 'main' into upgrade-jsforce-v3

* fixup! fixup! Merge branch 'main' into upgrade-jsforce-v3

---------

Co-authored-by: Alex van Andel <me@alexvanandel.com>
Co-authored-by: Morgan <33722304+ThyMinimalDev@users.noreply.github.com>
Co-authored-by: Morgan Vernay <morgan@cal.com>
Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

zomars committed:
chore: Webhook should timeout fast on e2e (#18570)

anikdhabal committed:
fix: wip (#18574)

MuhammadAimanSulaiman committed:
fix: resolve atoms build error

MuhammadAimanSulaiman committed:
fix: resolve atoms build error

hbjORbj committed:
cleanup: remove-headseo-from-migrated-app-router-components (#18576)

zomars committed:
chore: fix SVG warning on render (#18572)

anikdhabal committed:
fix: date filter for the same day showing appointments of the day before (#18579)

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

nizzyabi committed:
fix: billing plain chat working (#18530)

anikdhabal committed:
fix: unpublished org is visble as an option in workflow (#18522)

Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>

eunjae-lee committed:
fix: improve filters at /bookings (#18581)

* fix: improve filters at /bookings

* fix type

anikdhabal committed:
fix: flaky e2e test in saml.e2e (#18559)

* chore: skip failure e2e test

* Update .env

* Update manage-booking-questions.e2e.ts

* Update manage-booking-questions.e2e.ts

* Update manage-booking-questions.e2e.ts

* Update manage-booking-questions.e2e.ts

* Update manage-booking-questions.e2e.ts

* Update manage-booking-questions.e2e.ts

* Update .env

* Update manage-booking-questions.e2e.ts

* fix

* fix flake

* Update saml.e2e.ts

alishaz-polymath committed:
docs: fix footer socials in /help and other broken pages (#18578)

* Update mint.json

* fix broken links

---------

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

ibex088 committed:
fix: broken platform members table (#18583)

ibex088 committed:
feat: added pagination to platform managed users table (#18523)

* feat: added pagination to platform managed users table

* removed console

* implemented feedback, removed unused components

* added label

* Update managed-users-view.tsx

* Update managed-users-view.tsx

calcom-bot committed:
feat: update translations via @replexica (#18584)

Co-authored-by: Replexica <support@replexica.com>

ThyMinimalDev committed:
chore: improve ooo reason documentation apiv2 (#18582)

invalid-email-address committed:
chore: release v4.8.16

PeerRich committed:
feat: granola.ai app (#18585)

* new granola app

* granola app

kart1ka committed:
fix: wrong Meeting Ended payload in zapier (#18202)

* fix: wrong Meeting Ended payload in zapier

* fix

joeauyeung committed:
fix: Create dub links if valid url (#18588)

* Create dub links if valid

* Map through filtered links

---------

Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

alishaz-polymath committed:
docs: fix error introduced in #18578 (#18593)

* Fix error

* remove indent

---------

Co-authored-by: CarinaWolli <wollencarina@gmail.com>

sean-brydon committed:
feat: Round Robin weights future members toggle (#17782)

Co-authored-by: Omar López <zomars@me.com>

nizzyabi committed:
detailed customer card (#18511)

Co-authored-by: Omar López <zomars@me.com>

hbjORbj committed:
chore: app router - all sub-pages in `/apps` (#16976)

* chore: apps/[slug] remove pages router

* remove apps/[slug] pages from /future

* chore: apps/installed remove pages router

* chore: apps/installation remove pages router

* remove Head element

* fix metadata

* fix test

* fix another test

* chore: apps/categories remove pages router

* revert unneeded changes

* update middleware

* Remove <Head>

* remove unused import and code

* remove unused import and code again

* fix

* fix category page

* add split icon

* add /routing paths to middleware matcher

* wip

* remove HeadSeo from App.tsx

* clean up head-seo test

* add generateAppMetadata

* use generateAppMetadata in apps/[slug] page

* delete file

* remove log

* fix

* fix

* fix apps/installed pages

* fix cateogires pages

* fix

* fix imports

* wip

* fix

* fix

* fix metadata

* fix

* redirect /apps/routing-forms to /routing

* replace all usages of /apps/routing-forms to /routing

* better naming

* /routing -> /routing/forms

* fix

* fix

* fix

* fix

* remove backPath as it is irrelevant when withoutMain is true

* fix type checks

* fix type check in apps/[slug]

* refactors

* fix

* fix test

* fix

* fix

* fix

* Replace multiple leading slashes with a single slash

* migrate routing-forms too

* add re routing

* fix

* add redirection

---------

Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Anik Dhabal Babu <81948346+anikdhabal@users.noreply.github.com>

zomars committed:
chore: app router 404 page (#18597)

* wip

* wip

* fix not found page

* render middleware for /settings pages

* fix

* remove global-error page

* add metadata to not-found page

* make not-found page static

* remove 404

* adding not-found to middleware is not necessary

* add every routes to config.matcher

* fix test

* fix style

* use i18n string

* fix tests

* fix

* fix

* revert unneeded changes

* fix

* fix

* fix

* fix style

* fix

* remove 404

* remove log

* fix

* fix

* fix

* fix

* better naming

* parallel testing

---------

Co-authored-by: Benny Joo <sldisek783@gmail.com>

hbjORbj committed:
feat: render custom error page for unexpected sever error + remove` pages/_error` (#18606)

* remove page/_error

* refactor app/error

hbjORbj committed:
fix: app/not-found cannot be a static page (#18610)

sean-brydon committed:
Async False (#18611)

hbjORbj committed:
chore: redirect to /500 if pathname does not exist + better error handling (#18615)

* fix lint error

* fix booking page and better error handling

MuhammadAimanSulaiman committed:
test: added test drop-in urls

MuhammadAimanSulaiman committed:
feat: implemented dropin-ui using hitpay embed window.parent

MuhammadAimanSulaiman committed:
Merge branch 'hitpay-embed-10764' into hitpay-embed

MuhammadAimanSulaiman committed:
fix: increase validation limit to 128 in hitpay app setup

MuhammadAimanSulaiman committed:
refactor: update hitpay app description
